File: backend/src/main/java/edu/cit/daal/techtrack/dto/response/LoanResponse.java
package edu.cit.daal.techtrack.dto.response;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Builder
public class LoanResponse {

    private Long id;
    private BorrowerDto borrower;
    private AssetDto asset;
    private String purpose;
    private String status;
    private LocalDate requestedReturnDate;
    private Long approvedBy;
    private LocalDateTime approvedAt;
    private LocalDate actualReturnDate;
    private String conditionOnReturn;
    private String rejectionReason;
    private LocalDateTime requestedAt;

    @Data
    @Builder
    public static class BorrowerDto {
        private Long id;
        private String firstName;
        private String lastName;
        private String email;
        private String studentId;
        private String department;
    }

    @Data
    @Builder
    public static class AssetDto {
        private Long id;
        private String name;
        private String category;
        private String assetTag;
        private String status;
    }
}

File: Mobile/app/src/main/java/com/techtrack/inventory/data/remote/model/response/LoanResponse.kt
package com.techtrack.inventory.data.remote.model.response

data class LoanResponse(
    val id: Long,
    val asset: LoanAssetDto,
    val borrower: LoanBorrowerDto,
    val purpose: String,
    val status: String,
    val requestedReturnDate: String,
    val approvedBy: LoanBorrowerDto?,
    val approvedAt: String?,
    val actualReturnDate: String?,
    val conditionOnReturn: String?,
    val rejectionReason: String?,
    val requestedAt: String
)

data class LoanAssetDto(
    val id: Long,
    val name: String,
    val assetTag: String
)

data class LoanBorrowerDto(
    val id: Long,
    val firstName: String?,
    val lastName: String?
) {
    fun fullName() = listOfNotNull(firstName, lastName)
        .joinToString(" ")
        .ifBlank { "—" }
}
